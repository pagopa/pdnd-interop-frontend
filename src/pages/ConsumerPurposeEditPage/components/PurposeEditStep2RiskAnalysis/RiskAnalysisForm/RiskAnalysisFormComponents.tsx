import React from 'react'
import type { FormConfigQuestion } from '@/api/api.generatedTypes'
import type { Answers, Questions } from '../../../types/risk-analysis-form.types'
import { useFormContext } from 'react-hook-form'
import {
  RHFCheckboxGroup,
  RHFRadioGroup,
  RHFSelect,
  RHFTextField,
} from '@/components/shared/react-hook-form-inputs'
import { RiskAnalysisSwitch } from './components/RiskAnalysisSwitch'
import { useTranslation } from 'react-i18next'
import useCurrentLanguage from '@/hooks/useCurrentLanguage'
import {
  formatRiskAnalysisInputInfoLabel,
  formatRiskAnalysisInputLabel,
  formatRiskAnalysisInputValidationInfoLabel,
  getRiskAnalysisInputOptions,
} from '@/pages/ConsumerPurposeEditPage/utils/risk-analysis-form.utils'
import RiskAnalysisFormSection from './components/RiskAnalysisFormSection'

/**
 * Returns the updated form components.
 *
 * @param questions - the actual updated questions visible to the user
 * @returns Array of components that should be rendered inside the form
 * */
export const RiskAnalysisFormComponents: React.FC<{ questions: Questions }> = ({ questions }) => {
  const { t } = useTranslation('purpose')
  const lang = useCurrentLanguage()
  const answers = useFormContext<Answers>().watch()

  return React.useMemo(() => {
    function buildFormQuestionComponents(question: FormConfigQuestion) {
      const questionComponents: Array<React.ReactNode> = []

      const maxLength = question?.validation?.maxLength

      const inputOptions = getRiskAnalysisInputOptions(question, answers, lang)
      const label = formatRiskAnalysisInputLabel(question, lang, t)
      const infoLabel = formatRiskAnalysisInputInfoLabel(question, lang)

      const validationInfoLabel = formatRiskAnalysisInputValidationInfoLabel(question, t)

      const sx = { mb: 0 }
      const commonProps = {
        key: question.id,
        name: question.id,
        infoLabel: validationInfoLabel,
        sx,
      }

      switch (question.visualType) {
        case 'text':
          questionComponents.push(
            <RiskAnalysisFormSection
              title={label}
              description={infoLabel}
              formFieldName={question.id}
            >
              <RHFTextField
                {...commonProps}
                inputProps={{ maxLength }}
                rules={{ required: true }}
              />
            </RiskAnalysisFormSection>
          )
          break
        case 'select-one':
          questionComponents.push(
            <RiskAnalysisFormSection
              title={label}
              description={infoLabel}
              formFieldName={question.id}
            >
              <RHFSelect
                {...commonProps}
                options={inputOptions}
                emptyLabel={t('edit.step2.emptyLabel')}
                rules={{ required: true }}
              />
            </RiskAnalysisFormSection>
          )
          break
        case 'checkbox':
          questionComponents.push(
            <RiskAnalysisFormSection
              title={label}
              description={infoLabel}
              formFieldName={question.id}
            >
              <RHFCheckboxGroup
                {...commonProps}
                options={inputOptions}
                rules={{
                  validate: (value) =>
                    (typeof value !== 'undefined' && value.length > 0) ||
                    t('edit.step2.multiCheckboxField.validation.mixed.required'),
                }}
              />
            </RiskAnalysisFormSection>
          )
          break
        case 'radio':
          questionComponents.push(
            <RiskAnalysisFormSection
              title={label}
              description={infoLabel}
              formFieldName={question.id}
            >
              <RHFRadioGroup {...commonProps} options={inputOptions} rules={{ required: true }} />
            </RiskAnalysisFormSection>
          )
          break
        case 'switch':
          questionComponents.push(
            <RiskAnalysisFormSection
              title={label}
              description={infoLabel}
              formFieldName={question.id}
            >
              <RiskAnalysisSwitch
                {...commonProps}
                options={inputOptions}
                rules={{
                  validate: (value) =>
                    (typeof value === 'boolean' && value === true) ||
                    t('edit.step2.riskAnalysisSwitch.validation.boolean.isValue'),
                }}
              />
            </RiskAnalysisFormSection>
          )
          break
      }

      return questionComponents
    }

    const formComponents: Array<React.ReactNode> = []
    const questionIds = Object.keys(questions)

    questionIds.forEach((questionId) => {
      const question = questions[questionId]
      formComponents.push(...buildFormQuestionComponents(question))
    })

    return <>{formComponents}</>
  }, [lang, questions, t, answers])
}
