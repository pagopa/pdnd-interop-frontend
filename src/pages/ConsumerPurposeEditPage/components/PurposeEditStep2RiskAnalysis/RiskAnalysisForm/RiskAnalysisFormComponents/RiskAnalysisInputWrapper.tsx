import { SectionContainer } from '@/components/layout/containers'
import { FormControl, FormHelperText, FormLabel, Stack, Typography } from '@mui/material'
import React from 'react'

type RiskAnalysisInputWrapperProps = {
  children: React.ReactNode
  name?: string
  label: string
  infoLabel?: string
  helperText?: string
  error?: string
  labelId?: string
  infoLabelId?: string
  helperTextId?: string
  errorId?: string
  isInputGroup?: boolean
}

const RiskAnalysisInputWrapper: React.FC<RiskAnalysisInputWrapperProps> = ({
  children,
  name,
  label,
  infoLabel,
  helperText,
  helperTextId,
  labelId,
  infoLabelId,
  error,
  errorId,
  isInputGroup,
}) => {
  return (
    <SectionContainer component={isInputGroup ? 'fieldset' : 'div'}>
      <FormControl fullWidth error={!!error}>
        <Stack spacing={1} sx={{ mb: 4 }}>
          <FormLabel
            htmlFor={name}
            component={isInputGroup ? 'legend' : 'label'}
            id={labelId}
            sx={{ fontWeight: 600, maxWidth: 600 }}
          >
            {label}
          </FormLabel>
          {infoLabel && (
            <Typography id={infoLabelId} component="span" variant="caption" color="text.secondary">
              {infoLabel}
            </Typography>
          )}
        </Stack>
        {children}

        {helperText && (
          <FormHelperText
            component="span"
            id={helperTextId}
            error={false}
            sx={{ fontWeight: 400, ml: 0, display: 'block' }}
          >
            {helperText}
          </FormHelperText>
        )}

        {error && (
          <FormHelperText
            id={errorId}
            component="span"
            sx={{ fontWeight: 400, ml: 0, display: 'block' }}
          >
            {error}
          </FormHelperText>
        )}
      </FormControl>
    </SectionContainer>
  )
}

export default RiskAnalysisInputWrapper
