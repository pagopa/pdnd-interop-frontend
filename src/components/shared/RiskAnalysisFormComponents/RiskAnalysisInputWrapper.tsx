import { SectionContainer } from '@/components/layout/containers'
import { FormControl, FormHelperText, FormLabel, Stack, Typography } from '@mui/material'
import React from 'react'
import { RiskAnalysisAnswerComponent } from '@/pages/ConsumerPurposeTemplateEditPage/components/PurposeTemplateEditStepRiskAnalysis/RiskAnalysisForm/RiskAnalysisAnswerComponent'

type RiskAnalysisInputWrapperProps = {
  children: React.ReactNode
  name?: string
  label: string
  infoLabel?: string
  helperText?: string
  error?: string
  labelId?: string
  infoLabelId?: string
  helperTextId?: string
  errorId?: string
  isInputGroup?: boolean
  isFromPurposeTemplate?: boolean
}

const RiskAnalysisInputWrapper: React.FC<RiskAnalysisInputWrapperProps> = ({
  children,
  name,
  label,
  infoLabel,
  helperText,
  helperTextId,
  labelId,
  infoLabelId,
  error,
  errorId,
  isInputGroup,
  isFromPurposeTemplate,
}) => {
  return (
    <SectionContainer component={isInputGroup ? 'fieldset' : 'div'}>
      <SectionContainer
        innerSection
        {...(isFromPurposeTemplate ? { sx: { border: 1, borderColor: 'lightgrey', p: 3 } } : {})}
      >
        <FormControl fullWidth error={!!error}>
          <Stack spacing={1} sx={{ mb: 4 }}>
            <FormLabel
              htmlFor={name}
              component={isInputGroup ? 'legend' : 'label'}
              id={labelId}
              sx={{ fontWeight: 600 }}
            >
              {label}
            </FormLabel>
            {infoLabel && (
              <Typography
                id={infoLabelId}
                component="span"
                variant="caption"
                color="text.secondary"
              >
                {infoLabel}
              </Typography>
            )}
          </Stack>
          {children}

          {helperText && (
            <FormHelperText
              component="span"
              id={helperTextId}
              error={false}
              sx={{ fontWeight: 400, ml: 0, display: 'block' }}
            >
              {helperText}
            </FormHelperText>
          )}

          {error && (
            <FormHelperText
              id={errorId}
              component="span"
              sx={{ fontWeight: 400, ml: 0, display: 'block' }}
            >
              {error}
            </FormHelperText>
          )}
        </FormControl>
      </SectionContainer>
      {isFromPurposeTemplate && <RiskAnalysisAnswerComponent />}
    </SectionContainer>
  )
}

export default RiskAnalysisInputWrapper
