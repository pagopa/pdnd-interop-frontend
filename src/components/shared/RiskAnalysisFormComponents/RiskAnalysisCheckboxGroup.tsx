import React from 'react'
import { FormControlLabel, FormGroup, Checkbox } from '@mui/material'
import { Controller, useFormContext } from 'react-hook-form'
import type { InputOption } from '@/types/common.types'
import type { ControllerProps } from 'react-hook-form/dist/types'
import { useTranslation } from 'react-i18next'
import { mapValidationErrorMessages } from '@/utils/form.utils'
import RiskAnalysisInputWrapper from './RiskAnalysisInputWrapper'
import type { RiskAnalysisAnswers } from '@/types/risk-analysis-form.types'

export type RiskAnalysisCheckboxGroupProps = {
  questionId: string
  label: string
  infoLabel?: string
  helperText?: string
  options: Array<InputOption>
  rules?: ControllerProps['rules']
  isFromPurposeTemplate?: boolean
}

export const RiskAnalysisCheckboxGroup: React.FC<RiskAnalysisCheckboxGroupProps> = ({
  questionId,
  label,
  options,
  infoLabel,
  helperText,
  rules,
  isFromPurposeTemplate,
}) => {
  const { formState } = useFormContext<{ answers: RiskAnalysisAnswers }>()
  const { t } = useTranslation()

  const name = `answers.${questionId}`

  const error = formState.errors.answers?.[questionId]?.message as string | undefined

  return (
    <RiskAnalysisInputWrapper
      isInputGroup
      label={label}
      infoLabel={infoLabel}
      helperText={helperText}
      error={error}
      isFromPurposeTemplate={isFromPurposeTemplate}
      questionId={questionId}
    >
      <FormGroup>
        <Controller
          name={name}
          rules={mapValidationErrorMessages(rules, t)}
          render={({ field }) => {
            const onChange = (e: React.SyntheticEvent) => {
              const target = e.target as HTMLInputElement
              const prevValue = (field.value ?? []) as Array<string>

              const newValue = prevValue?.includes(target.name)
                ? prevValue.filter((v: unknown) => v !== target.name)
                : [...prevValue, target.name]

              field.onChange(newValue)
            }

            return (
              <>
                {options.map((o) => (
                  <FormControlLabel
                    key={o.value}
                    value={o.value}
                    control={
                      <Checkbox
                        checked={field.value?.includes(o.value) ?? false}
                        onChange={onChange}
                        name={String(o.value)}
                      />
                    }
                    label={o.label}
                    sx={{ mr: 3 }}
                  />
                ))}
              </>
            )
          }}
        />
      </FormGroup>
    </RiskAnalysisInputWrapper>
  )
}
