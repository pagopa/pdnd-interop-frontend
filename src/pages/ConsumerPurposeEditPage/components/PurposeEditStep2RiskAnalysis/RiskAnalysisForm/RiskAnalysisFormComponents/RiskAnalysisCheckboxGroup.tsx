import React from 'react'
import { FormControlLabel, FormGroup, Checkbox } from '@mui/material'
import { Controller, useFormContext } from 'react-hook-form'
import type { InputOption } from '@/types/common.types'
import type { ControllerProps } from 'react-hook-form/dist/types'
import { useTranslation } from 'react-i18next'
import { mapValidationErrorMessages } from '@/utils/form.utils'
import RiskAnalysisInputWrapper from './RiskAnalysisInputWrapper'

export type RiskAnalysisCheckboxGroupProps = {
  name: string
  label: string
  infoLabel?: string
  helperText?: string
  options: Array<InputOption>
  rules?: ControllerProps['rules']
}

export const RiskAnalysisCheckboxGroup: React.FC<RiskAnalysisCheckboxGroupProps> = ({
  name,
  label,
  options,
  infoLabel,
  helperText,
  rules,
}) => {
  const { formState } = useFormContext()
  const { t } = useTranslation()

  const error = formState.errors[name]?.message as string | undefined

  return (
    <RiskAnalysisInputWrapper
      isInputGroup
      label={label}
      infoLabel={infoLabel}
      helperText={helperText}
      error={error}
    >
      <FormGroup>
        <Controller
          name={name}
          rules={mapValidationErrorMessages(rules, t)}
          render={({ field }) => {
            const onChange = (e: React.SyntheticEvent) => {
              const target = e.target as HTMLInputElement
              const prevValue = (field.value ?? []) as Array<string>

              const newValue = prevValue?.includes(target.name)
                ? prevValue.filter((v: unknown) => v !== target.name)
                : [...prevValue, target.name]

              field.onChange(newValue)
            }

            return (
              <>
                {options.map((o) => (
                  <FormControlLabel
                    key={o.value}
                    value={o.value}
                    control={
                      <Checkbox
                        checked={field.value?.includes(o.value) ?? false}
                        onChange={onChange}
                        name={String(o.value)}
                      />
                    }
                    label={o.label}
                    sx={{ mr: 3 }}
                  />
                ))}
              </>
            )
          }}
        />
      </FormGroup>
    </RiskAnalysisInputWrapper>
  )
}
