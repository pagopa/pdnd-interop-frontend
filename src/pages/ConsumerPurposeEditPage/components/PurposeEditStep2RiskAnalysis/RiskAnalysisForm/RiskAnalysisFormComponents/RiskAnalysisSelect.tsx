import React from 'react'
import { MenuItem, Select as MUISelect, type SelectProps as MUISelectProps } from '@mui/material'
import { Controller, useFormContext } from 'react-hook-form'
import type { InputOption } from '@/types/common.types'
import type { ControllerProps } from 'react-hook-form/dist/types'
import { useTranslation } from 'react-i18next'
import { getAriaAccessibilityInputProps, mapValidationErrorMessages } from '@/utils/form.utils'
import RiskAnalysisInputWrapper from './RiskAnalysisInputWrapper'

export type RiskAnalysisSelectProps = Omit<MUISelectProps, 'onChange' | 'label'> & {
  name: string
  label: string
  infoLabel?: string
  helperText?: string
  emptyLabel?: string
  options: Array<InputOption>
  rules?: ControllerProps['rules']
}

export const RiskAnalysisSelect: React.FC<RiskAnalysisSelectProps> = ({
  name,
  label,
  options,
  infoLabel,
  helperText,
  emptyLabel,
  rules,
  ...props
}) => {
  const { t } = useTranslation()
  const { formState } = useFormContext()
  const error = formState.errors[name]?.message as string | undefined

  const { accessibilityProps, ids } = getAriaAccessibilityInputProps(name, {
    label,
    infoLabel,
    error,
    helperText,
  })

  return (
    <RiskAnalysisInputWrapper
      label={label}
      infoLabel={infoLabel}
      error={error}
      helperText={helperText}
      {...ids}
    >
      <Controller
        name={name}
        rules={mapValidationErrorMessages(rules, t)}
        render={({ field: { ref, onChange, ...fieldProps } }) => (
          <MUISelect
            {...props}
            {...fieldProps}
            inputProps={{
              ...props.inputProps,
              ...accessibilityProps,
            }}
            onChange={(e) => {
              onChange(e)
            }}
            inputRef={ref}
          >
            {options.length > 0 ? (
              options.map((o, i) => (
                <MenuItem key={i} value={o.value}>
                  {o.label}
                </MenuItem>
              ))
            ) : (
              <MenuItem value="">{emptyLabel ?? ''}</MenuItem>
            )}
          </MUISelect>
        )}
      />
    </RiskAnalysisInputWrapper>
  )
}
