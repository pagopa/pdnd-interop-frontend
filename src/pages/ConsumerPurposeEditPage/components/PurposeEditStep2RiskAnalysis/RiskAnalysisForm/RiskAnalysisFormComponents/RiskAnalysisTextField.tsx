import React from 'react'
import { OutlinedInput, type OutlinedInputProps } from '@mui/material'
import { useFormContext, Controller } from 'react-hook-form'
import type { ControllerProps } from 'react-hook-form/dist/types/controller'
import { getAriaAccessibilityInputProps, mapValidationErrorMessages } from '@/utils/form.utils'
import { useTranslation } from 'react-i18next'
import RiskAnalysisInputWrapper from './RiskAnalysisInputWrapper'

export type RiskAnalysisTextFieldProps = Omit<OutlinedInputProps, 'type'> & {
  name: string
  label: string
  infoLabel?: string
  helperText?: string
  formHelper?: string
  rules?: ControllerProps['rules']
}

export const RiskAnalysisTextField: React.FC<RiskAnalysisTextFieldProps> = ({
  name,
  label,
  infoLabel,
  helperText,
  multiline,
  rules,
  ...props
}) => {
  const { formState } = useFormContext()
  const { t } = useTranslation()

  const error = formState.errors[name]?.message as string | undefined

  const { accessibilityProps, ids } = getAriaAccessibilityInputProps(name, {
    label,
    infoLabel,
    error,
    helperText,
  })

  return (
    <RiskAnalysisInputWrapper
      name={name}
      label={label}
      infoLabel={infoLabel}
      helperText={helperText}
      error={error}
      {...ids}
    >
      <Controller
        name={name}
        rules={mapValidationErrorMessages(rules, t)}
        render={({ field: { ref, onChange, ...fieldProps } }) => (
          <OutlinedInput
            {...props}
            inputProps={{ ...props.inputProps, ...accessibilityProps }}
            multiline={multiline}
            rows={multiline ? 6 : undefined}
            error={!!error}
            onChange={(e) => {
              onChange(e)
            }}
            inputRef={ref}
            {...fieldProps}
          />
        )}
      />
    </RiskAnalysisInputWrapper>
  )
}
