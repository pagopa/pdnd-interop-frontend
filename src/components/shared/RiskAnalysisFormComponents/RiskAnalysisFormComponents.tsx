import React from 'react'
import type { FormConfigQuestion } from '@/api/api.generatedTypes'
import type { Answers, Questions } from './types/risk-analysis-form.types'
import { useFormContext } from 'react-hook-form'
import { useTranslation } from 'react-i18next'
import useCurrentLanguage from '@/hooks/useCurrentLanguage'
import {
  formatRiskAnalysisInputInfoLabel,
  formatRiskAnalysisInputLabel,
  formatRiskAnalysisHerlperText,
  getRiskAnalysisInputOptions,
} from '@/components/shared/RiskAnalysisFormComponents/utils/risk-analysis-form.utils'
import { RiskAnalysisSwitch } from './RiskAnalysisSwitch'
import { RiskAnalysisSelect } from './RiskAnalysisSelect'
import { RiskAnalysisTextField } from './RiskAnalysisTextField'
import { RiskAnalysisCheckboxGroup } from './RiskAnalysisCheckboxGroup'
import { RiskAnalysisRadioGroup } from './RiskAnalysisRadioGroup'

/**
 * Returns the updated form components.
 *
 * @param questions - the actual updated questions visible to the user
 * @returns Array of components that should be rendered inside the form
 * */
export const RiskAnalysisFormComponents: React.FC<{ questions: Questions }> = ({ questions }) => {
  const { t } = useTranslation('purpose')
  const lang = useCurrentLanguage()
  const answers = useFormContext<Answers>().watch()

  return React.useMemo(() => {
    function buildFormQuestionComponents(question: FormConfigQuestion) {
      const questionComponents: Array<React.ReactNode> = []

      const maxLength = question?.validation?.maxLength

      const inputOptions = getRiskAnalysisInputOptions(question, answers, lang)
      const label = formatRiskAnalysisInputLabel(question, lang, t)
      const infoLabel = formatRiskAnalysisInputInfoLabel(question, lang)

      const helperText = formatRiskAnalysisHerlperText(question, t)

      const sx = { mb: 0 }
      const commonProps = {
        key: question.id,
        name: question.id,
        id: question.id,
        label,
        infoLabel,
        helperText,
        sx,
      }

      switch (question.visualType) {
        case 'text':
          questionComponents.push(
            <RiskAnalysisTextField
              {...commonProps}
              inputProps={{ maxLength }}
              rules={{ required: true }}
            />
          )
          break
        case 'select-one':
          questionComponents.push(
            <RiskAnalysisSelect
              {...commonProps}
              options={inputOptions}
              emptyLabel={t('edit.step2.emptyLabel')}
              rules={{ required: true }}
            />
          )
          break
        case 'checkbox':
          questionComponents.push(
            <RiskAnalysisCheckboxGroup
              {...commonProps}
              options={inputOptions}
              rules={{
                validate: (value) =>
                  (typeof value !== 'undefined' && value.length > 0) ||
                  t('edit.step2.multiCheckboxField.validation.mixed.required'),
              }}
            />
          )
          break
        case 'radio':
          questionComponents.push(
            <RiskAnalysisRadioGroup
              {...commonProps}
              options={inputOptions}
              rules={{ required: true }}
            />
          )
          break
        case 'switch':
          questionComponents.push(
            <RiskAnalysisSwitch
              {...commonProps}
              options={inputOptions}
              rules={{
                validate: (value) =>
                  (typeof value === 'boolean' && value === true) ||
                  t('edit.step2.riskAnalysisSwitch.validation.boolean.isValue'),
              }}
            />
          )
          break
      }

      return questionComponents
    }

    const formComponents: Array<React.ReactNode> = []
    const questionIds = Object.keys(questions)

    questionIds.forEach((questionId) => {
      const question = questions[questionId]
      formComponents.push(...buildFormQuestionComponents(question))
    })

    return <>{formComponents}</>
  }, [lang, questions, t, answers])
}
