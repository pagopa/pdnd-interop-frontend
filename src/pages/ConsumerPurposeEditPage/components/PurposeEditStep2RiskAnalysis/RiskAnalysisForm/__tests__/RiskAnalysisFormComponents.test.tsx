import React from 'react'
import { FormProvider, useForm } from 'react-hook-form'
import { RiskAnalysisFormComponents } from '../RiskAnalysisFormComponents'
import { render } from '@testing-library/react'
import type { Questions } from '@/pages/ConsumerPurposeEditPage/types/risk-analysis-form.types'

const commonQuestionValues = {
  label: { it: 'test', en: 'test' },
  infoLabel: { it: 'test', en: 'test' },
  required: true,
  defaultValue: [],
  dependencies: [],
}

const questions: Questions = {
  '1': {
    dataType: 'SINGLE',
    visualType: 'text',
    id: '1',
    ...commonQuestionValues,
  },
  '2': {
    dataType: 'SINGLE',
    visualType: 'select-one',
    id: '2',
    ...commonQuestionValues,
  },
  '3': {
    dataType: 'SINGLE',
    visualType: 'radio',
    id: '3',
    ...commonQuestionValues,
  },
  '4': {
    dataType: 'SINGLE',
    visualType: 'switch',
    id: '4',
    ...commonQuestionValues,
  },
  '5': {
    dataType: 'MULTI',
    visualType: 'checkbox',
    id: '5',
    ...commonQuestionValues,
  },
}

describe('RiskAnalysisFormComponents', () => {
  it('should match the snapshot', () => {
    const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
      const formMethods = useForm()

      return <FormProvider {...formMethods}>{children}</FormProvider>
    }

    const { baseElement } = render(<RiskAnalysisFormComponents questions={questions} />, {
      wrapper: Wrapper,
    })
    expect(baseElement).toMatchSnapshot()
  })
})
