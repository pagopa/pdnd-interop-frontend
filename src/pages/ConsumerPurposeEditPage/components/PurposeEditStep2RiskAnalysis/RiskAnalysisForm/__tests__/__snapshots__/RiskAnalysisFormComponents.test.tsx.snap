// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RiskAnalysisFormComponents > should match the snapshot 1`] = `
<body>
  <div>
    <section
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-1okj3ks-MuiStack-root"
      />
      <div
        class="MuiBox-root css-zv7ju9"
      >
        <div
          class="css-1okj3ks-MuiStack-root"
        >
          <span
            class="MuiTypography-root MuiTypography-sidenav css-d2mesl-MuiTypography-root"
          >
            test (edit.step2.validation.required)
          </span>
          <span
            class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
          >
            test
          </span>
        </div>
        <div
          class="MuiBox-root css-1hhytss"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r0:"
                name="1"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-1okj3ks-MuiStack-root"
      />
      <div
        class="MuiBox-root css-zv7ju9"
      >
        <div
          class="css-1okj3ks-MuiStack-root"
        >
          <span
            class="MuiTypography-root MuiTypography-sidenav css-d2mesl-MuiTypography-root"
          >
            test (edit.step2.validation.required)
          </span>
          <span
            class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
          >
            test
          </span>
        </div>
        <div
          class="MuiBox-root css-1hhytss"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id=":r1:"
            />
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby=":r1: mui-component-select-2"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-component-select-2"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="2"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-1okj3ks-MuiStack-root"
      />
      <div
        class="MuiBox-root css-zv7ju9"
      >
        <div
          class="css-1okj3ks-MuiStack-root"
        >
          <span
            class="MuiTypography-root MuiTypography-sidenav css-d2mesl-MuiTypography-root"
          >
            test (edit.step2.validation.required)
          </span>
          <span
            class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
          >
            test
          </span>
        </div>
      </div>
    </section>
    <section
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-1okj3ks-MuiStack-root"
      />
      <div
        class="MuiBox-root css-zv7ju9"
      >
        <div
          class="css-1okj3ks-MuiStack-root"
        >
          <span
            class="MuiTypography-root MuiTypography-sidenav css-d2mesl-MuiTypography-root"
          >
            test (edit.step2.validation.required)
          </span>
          <span
            class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
          >
            test
          </span>
        </div>
        <div
          class="MuiBox-root css-1hhytss"
        >
          <label
            class="MuiFormLabel-root MuiFormLabel-colorPrimary css-l38uzw-MuiFormLabel-root"
          >
            <div
              class="css-1npm2ob-MuiStack-root"
            >
              <span
                class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
              >
                <span
                  class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
                >
                  <input
                    class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                    name="4"
                    type="checkbox"
                  />
                  <span
                    class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                />
              </span>
            </div>
          </label>
        </div>
      </div>
    </section>
    <section
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-1okj3ks-MuiStack-root"
      />
      <div
        class="MuiBox-root css-zv7ju9"
      >
        <div
          class="css-1okj3ks-MuiStack-root"
        >
          <span
            class="MuiTypography-root MuiTypography-sidenav css-d2mesl-MuiTypography-root"
          >
            test (edit.step2.validation.required, edit.step2.validation.multipleChoice)
          </span>
          <span
            class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
          >
            test
          </span>
        </div>
      </div>
    </section>
  </div>
</body>
`;
